modules/auxiliary/gather/cloud_lookup.rb

```diff
@@ -0,0 +1,706 @@
+
+    response = http_get_request_raw(
+      datastore['HOSTNAME'],
+      datastore['RPORT'],
+      datastore['SSL'],
+      nil,
+      datastore['URIPATH']
+    )
+    return nil if response.nil?
+
+    return detect_solution(response)
+  end
+
+  # ------------------------------------------------------------------------- #
+
+  def run
+    @my_action = pick_action
+    if @my_action.nil?
+      print_error("Couldn't determine the action automaticaly.")
+      return
+    end
+    vprint_status("Selected action: #{@my_action.name}")
+
+    print_status('Passive gathering information...')
+
+    domain_name = PublicSuffix.parse(datastore['HOSTNAME']).domain
+    ip_list = []
+
+    # ViewDNS.info
+    ip_records = grab_domain_ip_history(domain_name)
+    ip_list |= ip_records unless ip_records.eql? false
+    unless ip_records.eql? false
+      print_status(" * ViewDNS.info: #{ip_records.count.to_s} IP address found(s).")
+    end
+
+    # DNS Enumeration
+    if datastore['DNSENUM'].eql? true
+      ip_records = dns_enumeration(domain_name, datastore['THREADS'])
+      ip_list |= ip_records unless ip_records.eql? false
+      unless ip_records.eql? false
+        print_status(" * DNS Enumeration: #{ip_records.count.to_s} IP address found(s).")
+      end
+    end
+
+    # Censys search
+    if [datastore['CENSYS_UID'], datastore['CENSYS_SECRET']].none?(&:nil?)
+      ip_records = censys_search(domain_name, 'ipv4', datastore['CENSYS_UID'], datastore['CENSYS_SECRET'])
+      ip_list |= ip_records unless ip_records.eql? false
+      unless ip_records.eql? false
+        print_status(" * Censys IPv4: #{ip_records.count.to_s} IP address found(s).")
+        print_status
+      end
+    end
+
+    if ip_list.empty?
+      print_bad('No IP address found :-(')
+      return
+    end
+
+    # Cleaning IP addresses if nessesary
+    case @my_action.name
+    when /ArvanCloud/
+      ip_blacklist = get_arvancloud_ips
+    when /CloudFlare/
+      ip_blacklist = get_cloudflare_ips
+    when /CloudFront/
+      ip_blacklist = get_cloudfront_ips
+    when /Fastly/
+      ip_blacklist = get_fastly_ips
+    when /Incapsula/
+      ip_blacklist = get_incapsula_ips
+    when /InGen Security/
+      ip_list.uniq.each do | ip |
+        a = dns_get_a(ip.to_s)
+        ['binarysec', 'easywaf', 'ingensec', '127.0.0.1'].each do | signature |
+          ip_blacklist << ip.to_s if a.to_s.downcase.include? signature.downcase
+        end
+      end
+    when /Stackpath/
+      ip_blacklist = get_stackpath_ips
+    end
+
+    records = []
+    if ip_blacklist
+      print_status("Clean #{@my_action.name} server(s)...")
+      ip_list.uniq.each do | ip |
+        is_listed = false
+
+        ip_blacklist.each do | ip_range |
+          if IPAddr.new(ip_range).include? ip.to_s
+            is_listed = true
+            break
+          end
+        end
+
+        unless is_listed.eql? true
+          records << ip.to_s
+        end
+      end
```

Comment:
You don't really need to track `is_listed` at all, you can just add the ip when you find it.

```suggestion
        ip_blacklist.each do | ip_range |
          if IPAddr.new(ip_range).include? ip.to_s
            records << ip.to_s
            break
          end
        end
```

Reply:
Thanks, I need to exclude IP addresses that are blacklisted. And that adds the blacklisted IP addresses. I look at how I can make it cleaner.